app/src/pages/admin/projectsPage/projectName/projectName.jsx

```diff
@@ -24,46 +24,79 @@ import { ADMIN_PROJECTS_PAGE_EVENTS } from 'components/main/analytics/events';
 import { SCREEN_XS_MAX_MEDIA } from 'common/constants/screenSizeVariables';
 import { navigateToProjectAction } from 'controllers/administrate/projects';
 import { PROJECT_PAGE } from 'controllers/pages';
-import { assignedProjectsSelector } from 'controllers/user';
+import {
+  assignedProjectsSelector,
+  assignedOrganizationsSelector,
+  setActiveProjectKeyAction,
+  userRolesSelector,
+} from 'controllers/user';
+import { MANAGER, userRolesType } from 'common/constants/projectRoles';
+import { ADMINISTRATOR } from 'common/constants/accountRoles';
 import styles from './projectName.scss';
 
 const cx = classNames.bind(styles);
 
 @connect(
   (state, ownProps) => ({
-    isAssigned: !!assignedProjectsSelector(state)[ownProps.project.projectKey],
+    isProjectAssigned: !!assignedProjectsSelector(state)[ownProps.project.projectKey],
+    isOrganizationAssigned: Object.keys(assignedOrganizationsSelector(state)).some(
+      ({ organizationId }) => organizationId === ownProps.project.organizationId,
+    ),
+    userRoles: userRolesSelector(state),
   }),
   {
     navigateToProject: navigateToProjectAction,
+    setActiveProjectKey: setActiveProjectKeyAction,
   },
 )
 @track()
 export class ProjectName extends Component {
   static propTypes = {
     project: PropTypes.object.isRequired,
     navigateToProject: PropTypes.func.isRequired,
-    isAssigned: PropTypes.bool,
+    setActiveProjectKey: PropTypes.func.isRequired,
+    isOrganizationAssigned: PropTypes.bool,
+    isProjectAssigned: PropTypes.bool,
     tracking: PropTypes.shape({
       trackEvent: PropTypes.func,
       getTrackingData: PropTypes.func,
     }).isRequired,
     disableAnalytics: PropTypes.bool,
+    userRoles: userRolesType,
   };
 
   static defaultProps = {
-    isAssigned: false,
+    isOrganizationAssigned: false,
+    isProjectAssigned: false,
     disableAnalytics: false,
+    userRoles: {},
   };
 
   onProjectClick = (event) => {
-    const { tracking, isAssigned, disableAnalytics } = this.props;
-    if (!isAssigned && window.matchMedia(SCREEN_XS_MAX_MEDIA).matches) {
+    const {
+      tracking,
+      isOrganizationAssigned,
+      isProjectAssigned,
+      disableAnalytics,
+      project,
+      userRoles: { userRole, organizationRole },
+      setActiveProjectKey,
+      navigateToProject,
+    } = this.props;
+
+    const isAdmin = userRole === ADMINISTRATOR;
+    const isManager = organizationRole === MANAGER;
+    const isNotPermission =
+      !isProjectAssigned && (!isAdmin || (isManager && isOrganizationAssigned));
```

Comment:
All of this stuff together with values from selectors can be encapsulated within Redux action and corresponding saga.

Reply:
Done.